addons/mail/static/src/messaging/entity/message/message.js

```diff
@@ -235,196 +326,162 @@ function MessageFactory({ Entity }) {
+                const originalBody = this.body;
+                prettyBody = this.body.replace(
+                    regexp,
+                    ` <span class="o_mail_emoji">${unicode}</span> `
                 );
-                for (const channel of oldPrevChannels) {
-                    for (const cache of channel.caches) {
-                        this.unlink({ threadCaches: cache });
-                    }
+                // Idiot-proof limit. If the user had the amazing idea of
+                // copy-pasting thousands of emojis, the image rendering can lead
+                // to memory overflow errors on some browsers (e.g. Chrome). Set an
+                // arbitrary limit to 200 from which we simply don't replace them
+                // (anyway, they are already replaced by the unicode counterpart).
+                if (_.str.count(prettyBody, "o_mail_emoji") > 200) {
+                    prettyBody = originalBody;
                 }
             }
+            // add anchor tags to urls
+            return parseAndTransform(prettyBody, addLink);
+        }
+
+        /**
+         * @override
+         */
+        _createInstanceLocalId(data) {
+            return `${this.constructor.entityName}_${data.id}`;
         }
 
     }
 
+    Message.entityName = 'Message';
+
     Message.fields = {
+        allThreads: many2many('Thread', {
+            compute: '_computeAllThreads',
+            dependencies: [
+                'originThread',
+                'threadCachesThread',
+            ],
+        }),
         attachments: many2many('Attachment', {
             inverse: 'messages',
         }),
         author: many2one('Partner'),
+        body: attr({
+            default: "",
+        }),
         checkedThreadCaches: many2many('ThreadCache', {
             inverse: 'checkedMessages',
         }),
+        customer_email_data: attr(),
+        customer_email_status: attr(),
+        date: attr({
+            default: moment(),
+        }),
+        email_from: attr(),
+        hasCheckbox: attr({
+            compute: '_computeHasCheckbox',
+            default: false,
+            dependencies: ['isModeratedByUser'],
+        }),
+        id: attr(),
+        isModeratedByUser: attr({
+            compute: '_computeIsModeratedByUser',
+            default: false,
+            dependencies: [
+                'moderation_status',
+                'originThread',
+                'originThreadIsModeratedByUser',
+            ],
+        }),
+        isTemporary: attr({
+            default: false,
+        }),
+        isTransient: attr({
+            default: false,
+        }),
+        is_discussion: attr({
+            default: false,
+        }),
+        is_note: attr({
+            default: false,
+        }),
+        is_notification: attr({
+            default: false,
+        }),
+        model: attr({
+            default: 'mail.message',
+        }),
```

Comment:
What is this?
Either I don't understand why it defaults to mail.message, or I don't understand why I could take any other value.